server/src/main/java/org/elasticsearch/index/query/SearchExecutionContext.java

```diff
@@ -722,4 +724,16 @@ public void setRewriteToNamedQueries() {
     public boolean rewriteToNamedQuery() {
         return rewriteToNamedQueries;
     }
+
+    public void explain(boolean explain) {
+        this.explain = explain;
```

Comment:
we don't need all this. All this logic is already handled, we just need to take advantage of it. Let's remove all these context updates.